testing/test_basic_api.py

```diff
@@ -109,9 +109,10 @@ def test_dump_version(tmpdir):
     expected_version_tuple_line = 'version_tuple = (1, 0, "dev42")'
     assert expected_version_tuple_line in lines
 
-    version_tuple = ""  # If exec() fails, the test will fail, as str is not tuple
-    exec(lines[lines.index(expected_version_tuple_line)])
-    assert isinstance(version_tuple, tuple)
+    # Assert that the version_tuple is actually a tuple
+    exec_locals = {}
```

Comment:
that's absolutely the wrong test,
the input type to the format sting has to be a tuple instead of a string,
for the issue at hand it is absolutely irrelevant what the output is . the type being passed to the formatting of the template has to be set correct